mlir/utils/jenkins/Jenkinsfile

```diff
@@ -386,7 +386,6 @@ pipeline {
                             LD_LIBRARY_PATH="${WORKSPACE}/MIOpen/build/lib:$LD_LIBRARY_PATH:"
                         }
                         steps {
-                            sh 'rm build/CMakeCache.txt'
```

Comment:
From the CI, it seems that this does not reduce the number of objects needed to be built for the static library. Not sure about the other way around. You may want to try a nightly to see how it goes.

Reply:
What I think might be happening here is that some of those flags are changing the CXXFLAGS, which would be causing the mass recompile. I might want to

1. Change that if/elseif into a three-way if/elseif/else (use PR defaults, no forcing)
2. Check if we actually need to force all those flags. My very preliminary suspect here is LLVM_BUILD_EXAMPLES